src/main/java/com/commercetools/sync/customers/helpers/CustomerBatchValidator.java

```diff
@@ -0,0 +1,223 @@
+        if (!indexesOfDuplicateAddressKeys.isEmpty()) {
+            handleError(format(ADDRESSES_THAT_KEYS_NOT_UNIQUE, customerDraft.getKey(), indexesOfDuplicateAddressKeys));
+            return false;
+        }
+
+        return true;
+    }
+
+    @Nonnull
+    private List<Integer> getIndexes(@Nonnull final List<Address> list, @Nonnull final Predicate<Address> predicate) {
+
+        final List<Integer> indexes = new ArrayList<>();
+        for (int i = 0; i < list.size(); i++) {
+            if (predicate.test(list.get(i))) {
+                indexes.add(i);
+            }
+        }
+        return indexes;
+    }
+
+    private boolean hasValidBillingAndShippingAddresses(@Nonnull final CustomerDraft customerDraft) {
+        /* An example error response from the API, when the indexes are not valid:
+           {
+                "statusCode": 400,
+                "message": "Customer does not contain an address at index '1'.",
+                "errors": [{
+                    "code": "InvalidOperation",
+                    "message": "Customer does not contain an address at index '-1'."
+                },
+                {
+                    "code": "InvalidOperation",
+                    "message": "Customer does not contain an address at index '1'."
+                }]
+            }
+         */
+
+        final List<Address> addressList = customerDraft.getAddresses();
+        final Predicate<Integer> isInvalidIndex = index -> index == null || addressList == null
+            || addressList.isEmpty() || index < 0 || index > addressList.size() - 1;
+
+        if (customerDraft.getBillingAddresses() != null && !customerDraft.getBillingAddresses().isEmpty()) {
+            final List<Integer> invalidIndexes = getIndexValues(customerDraft.getBillingAddresses(), isInvalidIndex);
+            if (!invalidIndexes.isEmpty()) {
+                handleError(format(BILLING_ADDRESSES_ARE_NOT_VALID, customerDraft.getKey(), invalidIndexes));
+                return false;
+            }
+        }
+
+        if (customerDraft.getShippingAddresses() != null && !customerDraft.getShippingAddresses().isEmpty()) {
+            final List<Integer> invalidIndexes = getIndexValues(customerDraft.getShippingAddresses(), isInvalidIndex);
+            if (!invalidIndexes.isEmpty()) {
+                handleError(format(SHIPPING_ADDRESSES_ARE_NOT_VALID, customerDraft.getKey(), invalidIndexes));
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    @Nonnull
+    private List<Integer> getIndexValues(@Nonnull final List<Integer> list,
+        @Nonnull final Predicate<Integer> predicate) {
+
+        final List<Integer> indexes = new ArrayList<>();
+        for (Integer integer : list) {
+            if (predicate.test(integer)) {
+                indexes.add(integer);
+            }
+        }
+        return indexes;
```

Comment:
```suggestion
return list.stream().filter(predicate).collect(Collectors.toList());
```

Reply:
same here, don't think your one-liner will do the same.